discovery/trends/v0.1/trends.md

```diff
@@ -0,0 +1,355 @@
+
+Uniquely identifying hashtags and links is more complex.
+
+Hashtags are represented by their name, a UTF-8 encoded string beginning
+with `#`. Different hashtag names might refer to the same concept,
+especially if they only differ in their use of upper and lower case
+characters. For example `#activitypub` and `#ActivityPub` can be
+considered to be the same.
+
+Links are HTTP(S) URLs. Syntactically different URLs can refer to the same
+resource. A simple example is again case-sensitivity as the domain part
+of an HTTP(S) URL is case-insensitive. So `https://example.com/test` and
+`https://EXAMPLE.com/test` both refer to the same resource.
+
+Fediverse servers already have to deal with this problem and will in
+most cases have some concept of "normalization" for both hashtags and
+links. This "normalization" is not part of any standard and different
+fediverse software can have different and even incompatible
+implementations of this.
+
+That is why when sharing trending hashtags and links with fediverse
+servers, FASP do not need to take special care how to represent them and
+can freely choose the representation (e.g. the first one they
+encountered or a normalized version). Fediverse servers SHOULD handle
+them just like they handle these representations in other contexts.
+
+When computing trends however, FASP SHOULD consider normalization to
+minimize the risk of obvious duplicates.
+
+For hashtags possible strategies for this besides case-insensitive
+handling of names include "ASCII folding" and using existing character
+transliteration rules. As all of these approaches have their downsides
+and can lead to two different words being erroneously treated as the
+same it is up to FASP to decide which strategy to use.
+
+For links [RFC-3986](https://tools.ietf.org/html/rfc3986.html) describes
+different normalization techniques in section 6.
+
+## Availability of Content for Hashtags and Links
+
+When a fediverse server requests trending hashtags or links it will in
+most cases want to display them to users and have a way for users to see
+how and where those have been used. Since FASP have a wider view of the
+fediverse it might be the case that the fediverse server has no content
+at all that references a trending hashtag or link. Or it might not have
+recent content that does so.
+
+To help fediverse servers in this situation discovery FASP include a
+couple of example URIs of content with every result of a query for
+hashtags and links trends. This allows fediverse servers to quickly fill
+their local caches with some content for the received trends. See the
+sections "Requesting Trending Hashtags" and "Requesting Trending Links"
+below for details.
+
+## Requesting Trends
+
+### Common Request Options
+
+All requests for trends from a fediverse server to FASP are HTTP `GET`
+requests that allow - with one exception - using the same optional
+parameters:
+
+* `withinLastHours`: This MUST be a positive integer specifying the
```

Comment:
How is can this be rounded to actually available time periods? For example, my FASP holds trends for 4-hour, 1-day, and 1-week intervals in precomputed materialized views.

Consequently, the actual time period should be indicated in the response.

Reply:
Good question. I am not sure if this fine-grained option to pass an arbitrary number of hours is even necessary. Maybe we could come up with some pre-defined timeframes (e.g. last hour, last 6, 12, 24, 48 and one week).

My only goal here was to offer something that is more flexible than what we currently have in Mastodon.

I would love to hear other people's thought on this.

Reply:
Maybe a Trend FASP could advertise which granularity and retention periods it supports?